spec/unit/coinbase/address/external_address_spec.rb

```diff
@@ -204,4 +291,78 @@
       end
     end
   end
+
+  shared_examples 'it called staking context balances' do
+    it 'creates a new staking_api_client' do
+      subject
+
+      expect(Coinbase::Client::StakeApi).to have_received(:new)
+    end
+
+    it 'calls staking_context' do
+      subject
+
+      expect(stake_api).to have_received(:get_staking_context).with(
+        asset_id: eth_asset.asset_id,
+        address_id: id,
+        network_id: 'ethereum-mainnet',
+        options: {
+          mode: mode
+        }
+      )
+    end
+  end
+
+  describe '#staking_balances' do
+    subject { address.staking_balances(eth_asset.asset_id, mode: mode) }
+
+    it 'returns the staking balances' do
+      expect(subject).to eq(
+        stakeable_balance: stake_balance,
+        unstakeable_balance: unstake_balance,
+        claimable_balance: claim_stake_balance
+      )
+    end
+
+    it_behaves_like 'it called staking context balances'
+  end
+
+  describe '#stakeable_balance' do
+    subject { address.stakeable_balance(eth_asset.asset_id, mode: mode) }
+
+    it_behaves_like 'it called staking context balances'
+
+    it { is_expected.to eq(stake_balance) }
+  end
+
+  describe '#unstakeable_balance' do
+    subject { address.unstakeable_balance(eth_asset.asset_id, mode: mode) }
+
+    it_behaves_like 'it called staking context balances'
+
+    it { is_expected.to eq(unstake_balance) }
+  end
+
+  describe '#claimable_balance' do
+    subject { address.claimable_balance(eth_asset.asset_id, mode: mode) }
+
+    it_behaves_like 'it called staking context balances'
+
+    it { is_expected.to eq(claim_stake_balance) }
+  end
+
+  describe '#staking_rewards' do
```

Comment:
Would be nice to separately test setting `format` as well as start/end defaults by stubbing datetime / using Timecop.

But lets do that separately